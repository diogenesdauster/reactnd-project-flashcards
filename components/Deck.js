import React, { PureComponent } from "react";
import { View, Text, TouchableWithoutFeedback, StyleSheet } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import PropTypes from "prop-types";
import { white, gray, yellow } from "../utils/colors";

class Deck extends PureComponent {
  static propTypes = {
    cardsCount: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    navigation: PropTypes.shape({ navigate: PropTypes.func.isRequired })
      .isRequired
  };

  state = {
    color: white
  };

  _onPress = () => {
    const { navigation, title } = this.props;
    navigation.navigate("DeckDetail", { title });
  };
  _onPressIn = () => {
    this.setState({
      color: yellow
    });
  };

  _onPressOut = () => {
    this.setState({
      color: white
    });
  };

  render() {
    const { title, cardsCount } = this.props;
    const { color } = this.state;
    return (
      <TouchableWithoutFeedback
        onPress={this._onPress}
        onPressIn={this._onPressIn}
        onPressOut={this._onPressOut}
        delayPressOut={500}
      >
        <View style={styles.container}>
          <View style={styles.subHeaderContent}>
            <MaterialCommunityIcons name="cards" size={20} color={gray} />
            <Text style={[styles.divider, styles.color]}>|</Text>
            <Text style={styles.color}>{cardsCount} Cards</Text>
          </View>
          <Text style={styles.header}>{title}</Text>
          <View style={{ flex: 1, backgroundColor: color, height: 4 }} />
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginLeft: 30,
    marginRight: 30,
    marginTop: 17,
    justifyContent: "center",
    padding: 20,
    backgroundColor: "#fff",
    shadowRadius: 3,
    shadowOpacity: 0.8,
    shadowColor: "rgba(0, 0, 0, 0.24)",
    shadowOffset: {
      width: 0,
      height: 3
    }
  },
  header: {
    fontSize: 20,
    paddingTop: 5
  },
  subHeaderContent: {
    flexDirection: "row"
  },
  color: {
    color: gray
  },
  divider: {
    paddingLeft: 5,
    paddingRight: 5
  }
});

export default Deck;
