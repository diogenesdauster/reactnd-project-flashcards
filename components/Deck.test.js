import "react-native";
import React from "react";
import Deck from "./Deck";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

// static propTypes = {
//     cardsCount: PropTypes.number.isRequired,
//     title: PropTypes.string.isRequired,
//     navigation: PropTypes.shape({ navigate: PropTypes.func.isRequired })
//       .isRequired
//   };

it("renders Deck correctly", () => {
  const mockNavigation = { navigate: jest.fn() };
  const tree = renderer
    .create(
      <Deck cardsCount={1} title="CardTest" navigation={mockNavigation} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
