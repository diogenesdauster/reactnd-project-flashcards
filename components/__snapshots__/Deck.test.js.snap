// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Deck correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "justifyContent": "center",
      "marginLeft": 30,
      "marginRight": 30,
      "marginTop": 17,
      "padding": 20,
      "shadowColor": "rgba(0, 0, 0, 0.24)",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 3,
    }
  }
  testID={undefined}
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "paddingLeft": 5,
            "paddingRight": 5,
          },
          Object {
            "color": "#757575",
          },
        ]
      }
    >
      |
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#757575",
        }
      }
    >
      1
       Cards
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "paddingTop": 5,
      }
    }
  >
    CardTest
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "height": 4,
      }
    }
  />
</View>
`;
